File: C++/008 Sorting/selection-sort.cpp
#include <bits/stdc++.h>
using namespace std;

/*
    Time Complexity - ~O(n^2)
*/

void selection_sort(int arr[], int n)
{
    for (int i = 0; i <= n - 2; i++)
    {
        int mini = i;
        for (int j = i; j <= n - 1; j++)
        {
            if (arr[j] < arr[mini])
            {
                mini = j;
            }
        }
        int temp = arr[mini];
        arr[mini] = arr[i];
        arr[i] = temp;
    }
}
int main()
{
    int arr[] = {13, 46, 24, 52, 20, 9};
    int n = sizeof(arr) / sizeof(arr[0]);
    selection_sort(arr, n);
    for (auto it : arr)
    {
        cout << it << " ";
    }
    return 0;
}

File: JavaScript/002 Sorting/selection-sort.js
/*
    Time complexity: O(N2), (where N = size of the array), for the best, worst, and average cases.
    Space Complexity - O(1)
*/

function selectionSort(arr) {
    const n = arr.length;

    // Selection sort
    for (let i = 0; i < n - 1; i++) {
        let mini = i;
        for (let j = i + 1; j < n; j++) {
            if (arr[j] < arr[mini]) {
                mini = j;
            }
        }
        // Swap elements
        let temp = arr[mini];
        arr[mini] = arr[i];
        arr[i] = temp;
    }

    console.log("After selection sort:");
    console.log(arr.join(" "));
}

const arr = [13, 46, 24, 52, 20, 9];

console.log("Before selection sort:");
console.log(arr.join(" "));
selectionSort(arr);
